### ISSUE TITLE
wrong vhost from virtual with side effect

### ISSUE THREAD

**Issue by auguster** [tomav/docker-mailserver#121]
I'm trying to configure postfix to accept multiple domains, so that say all map to the same corresponding mailboxes. For example, I want the mails for userA@domainB.tld and userA@domainA.tld to go into the same mailbox. I haven't find how to do that yet.

Anyway, during my tests I noticed there is something wrong with the way vhost is generated from virtual. In the following example the server's hostname is mail.domainA.tld the given path are from within the docker container:

/etc/postfix/virtual:
```userA@domainB.tld userA@domainA.tld```

/etc/postfix/vhost:
```
domainA.tld
domainB.tld userA
```

This has the side effect of causing opendkim to regenerate the key for domainB.tld at each launch. This is due to the fact that start-mailserver.sh is looking for the file `/etc/opendkim/keys/domainB.tld userA/mail.private`, doesn't find it, ask opendkim to generate one for "domainB.tld userA". Opendkim uses only the "domainB.tld" argument and so overwrites the previous key for "domainB.tld".

**Comment by tomav** [comment]
Perhaps @bilak may help on this.

**Comment by bilak** [comment]
Hi, try to mount you opendkim configuration from volume and it should be ok.

**Comment by tomav** [comment]
Hmm, the problem is that `domainB.tld userA` is wrong. It should only be the domain name.

@auguster could you post a link to a repo that we can clone and start to see what happens?

Thank you.

**Comment by auguster** [comment]
@bilak I already did that, the issue I describe overwrites the existing files anyway.
@tomav you want me to commit the machine and post the docker image somewhere or you just want a copy of /etc/postix ?

Normally you can easily reproduce the issue:
accounts.cf
```
userA@domainA.tld|password
```
for the two other files, the content is in my initial post.

**Comment by tomav** [comment]
I'll try with the information provided in this post.

**Comment by tomav** [comment]
Just tried from latest image and here what I have:

```
➜  docker-mailserver git:(master) ✗ docker-compose up -d
Creating dockermailserver_mail_1

➜  docker-mailserver git:(master) ✗ docker exec dockermailserver_mail_1 cat /tmp/postfix/accounts.cf
userA@domainA.tld|password

➜  docker-mailserver git:(master) ✗ docker exec dockermailserver_mail_1 cat /tmp/postfix/virtual
userA@domainB.tld userA@domainA.tld

➜  docker-mailserver git:(master) ✗ docker exec dockermailserver_mail_1 cat /etc/postfix/vhost
domainA.tld
domainB.tld
```

The `vhost` file is correct.

**Comment by auguster** [comment]
That's normal, the output you show is from `/etc/postfix` and not `/tmp/postfix`. The latter is not modified by `start-mailserver.sh`.

I did not use docker-compose (as it's not compatible with fleetctl on CoreOS) so my hostname is   `mail.domaina.tld`, but I don't suppose it's important.

**Comment by bilak** [comment]
@auguster how are you mounting the dkim? Try to look at [this line](https://github.com/tomav/docker-mailserver/blob/master/start-mailserver.sh#L68) and decide if you have correctly mounted the dkim folder.

**Comment by auguster** [comment]
@bilak `-v /data/data/mail/dkim:/etc/opendkim/` it's not a problem of persistence of the file but a problem of having those file overwritten.

**Comment by tomav** [comment]
@auguster 2 first output are input files.
The last one is output file from `/etc/postfix`

**Comment by auguster** [comment]
oops, sorry, my bad (friday effect)...
I'll try to make it into an automated script to make it easier to reproduce.

**Comment by bilak** [comment]
@auguster maybe I don't get the issue here. Are you telling me that those files are overwritten even if tehre exists file `/etc/opendkim/keys/$domainname/mail.private` for specified domain? I've tested that (not within image) and I don't see the issue.

**Comment by tomav** [comment]
@auguster Try with this working example.

**Comment by auguster** [comment]
@bilak exactly, I end up with `$domainname` containing "domainB.tld userA" because of the issue in the last line of my vhost.
@tomav ok, I will

**Comment by 00angus** [comment]
@auguster Sorry just a question to better understand the issue.
You have configured a mailserver to handle email for domainA (accounts.cf), but you want email for domainB (not managed by your server) to be redirected ? I'm probably missing smtg of your setup :)

**Comment by auguster** [comment]
@00angus the mailserver handles email for `domainA` as you said and I like it to also handle email for `domainB` that would be an alias for `domainB`. mail.domaina.tld and mail.domainb.tld are both pointing to the same server.

So I'd like mail for the same username on both domains to end up in the same mailbox. I'd be better if it were domainA for historic reasons but its not a strong constraint.

**Comment by 00angus** [comment]
@auguster ok - thank you for the clarification.
Now... if I understand correctly the problem is with the incorrect generation of "`vhost`" file: the opendkim issue is a by product of it.
The startup script [here](https://github.com/tomav/docker-mailserver/blob/master/start-mailserver.sh#L50) iterate on the content of the virtual file and your line, from your `virtual` example:

      virtual:
      userA@domainB.tld userA@domainA.tld

would output to vhost : `domainB.tld`. I've just tested this behaviour.
So when you write that the content of your vhost is something like:

      vhost:
      domainB.tld userA

I'm a little bit confused. I don't get how : 

      domain=$(echo ${from} | cut -d @ -f2)

can output your line! Does for example your `virtual` file contains something else ?

**Comment by auguster** [comment]
I found out what was wrong, in my `virtual` file, I separated the source and the destination column with a tabulation instead of a space. Because the IFS is set to space on [L47 of start-mailserver.sh](https://github.com/tomav/docker-mailserver/blob/master/start-mailserver.sh#L47) it considers the line as a single column.

With one or more spaces it works fine ;)

I'm not sure specifying IFS is useful for this specific while, I'd even think it's more robust without it.
